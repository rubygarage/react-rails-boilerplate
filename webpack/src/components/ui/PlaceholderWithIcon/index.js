import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'components/ui/Icon'

function PlaceholderWithIcon({ iconName, className, text }) {
  return (
    <span>
      <Icon type={iconName} className={className} />
      {text}
    </span>
  )
}

PlaceholderWithIcon.propTypes = {
  className: PropTypes.string,
  iconName: PropTypes.string,
  text: PropTypes.string
}

export default PlaceholderWithIcon
