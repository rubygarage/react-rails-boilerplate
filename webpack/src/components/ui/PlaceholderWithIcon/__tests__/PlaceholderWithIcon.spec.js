import React from 'react'
import { shallow } from 'enzyme'
import PlaceholderWithIcon from 'components/ui/PlaceholderWithIcon'

describe('<PlaceholderWithIcon />', () => {
  const defaultProps = {
    className: 'mr-10',
    iconName: 'icon-audio-circle',
    text: 'Select...'
  }

  it('renders placeholder with icon', () => {
    const component = shallow(
      <PlaceholderWithIcon {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })
})
