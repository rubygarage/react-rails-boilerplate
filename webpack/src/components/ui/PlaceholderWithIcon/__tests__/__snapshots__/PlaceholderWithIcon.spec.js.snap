// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceholderWithIcon /> renders placeholder with icon 1`] = `
<span>
  <Icon
    className="mr-10"
    type="icon-audio-circle"
  />
  Select...
</span>
`;
